src/main/java/seedu/address/logic/commands/DeleteSupplierCommand.java

```diff
@@ -0,0 +1,42 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Deletes a supplier identified using it's displayed index from the address book.
+ */
+public class DeleteSupplierCommand extends Command {
```

Comment:
the logic looks good. it may be ideal to edit the existing DeleteCommand class instead though as we can simply rename the file in the subsequent iterations and it will be applied across all of its usages.